src/natcap/invest/datastack.py

```diff
@@ -325,7 +339,7 @@ def _recurse(args_param, handler, nested_key=None):
                 handler.removeFilter(args_key_filter)
             return new_dict
         elif isinstance(args_param, list):
-            return [_recurse(list_item, handler) for list_item in args_param]
+            return [_recurse(list_item, handler, nested_key=nested_key) for list_item in args_param]
```

Comment:
Without `nested_key=nested_key`, `nested_key` would get reset to `None` anytime there's a list. That's a problem for the folder-naming, so I added this. It will affect the log filtering as well, but I don't think that's a problem... any idea why we would want the log filter to be reset when there's a list?

The other issue here is that, if there is a list, there is no longer a unique prefix for each folder. Each path in the list would go into a new folder with the same prefix. The only way to identify them separately would be to incorporate something from the path, such as the filename, into the folder name. Since I was previously using only the filename as the prefix, and not the nested_key, if this is needed I would be in favor of just naming folders according to filenames.

Reply:
Hey @emlys I think I'm finally wrapping my head around what Datastack is trying to do when archiving. 

For the list issue, is it possible to pass that `nested_key` like you have, but tack on the index of the value being passed to the `nested_key`? Something like:
```python
return [_recurse(list_item, handler, nested_key=f"{nested_key}{idx}") for idx, list_item in enumerate(args_param)]
```
I'm trying to think of how that could fail...?